### PR TITLE
tables: check handle consistency

### PR DESCRIPTION
**Pull request by ekexium**
Signed-off-by: ekexium <ekexium@gmail.com>

### What problem does this PR solve?

Issue Number: https://github.com/pingcap/tidb/issues/26833

Problem Summary:

Checks that handles in `row.key` and `index.value` are consistent.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
'None'
```

### CODE REVIEW

File: table/tables/mutation_checker.go [hunk 1]
```diff
@@ -36,6 +36,11 @@ type mutation struct {
 	value []byte
 }
 
+type richIndexMutation struct {
+	mutation
+	indexID int64
```
Comment (cfzjywxk) [inline table/tables/mutation_checker.go:41]: Could this filed be saved in the `mutation` so we don't need to build `richIndexMutation` which brings extra cost.

File: table/tables/mutation_checker.go [hunk 2]
```diff
@@ -47,20 +52,14 @@ type columnMaps struct {
...
-//
-// Assume the set of row values changes from V1 to V2, we check
-// (1) V2 - V1 = {added indices}
-// (2) V1 - V2 = {deleted indices}
-//
-// To check (1), we need
-// (a) {added indices} is a subset of {needed indices} => each index mutation is consistent with the input/row key/value
-// (b) {needed indices} is a subset of {added indices}. The check process would be exactly the same with how we generate
-// 		the mutations, thus ignored.
 func CheckIndexConsistency(
```
Comment (cfzjywxk) [inline table/tables/mutation_checker.go:63]: For the `CheckIndexConsistency` and `checkRowInsertionConsistency` we could add some comments explaining the theoretical performance overhead such as looping though array and allocating objects.

File: table/tables/mutation_checker_test.go [hunk 3]
```diff
@@ -262,42 +265,66 @@ func TestCheckIndexKeys(t *testing.T) {
...
 						maps.IndexIDToRowColInfos,
 					)
 					require.Nil(t, err)
+
+					// test checkHandleConsistency
+					rowKey := tablecodec.EncodeRowKeyWithHandle(table.tableID, handle)
+					rowValue, err := tablecodec.EncodeRow(sessVars.StmtCtx, rowToInsert, []int64{1, 2}, nil, nil, &rd)
+					require.Nil(t, err)
+					rowMutation := mutation{key: rowKey, value: rowValue}
+					err = checkHandleConsistency(rowMutation, richIndexMutations, maps.IndexIDToInfo)
```
Comment (cfzjywxk) [inline table/tables/mutation_checker_test.go:304]: May need to add some tests returning expected error.

**Comment by ekexium** [comment]
run-check_dev

**Pull request merged**